### PR TITLE
action 6440 - canceling reactions

### PR DESCRIPTION
**Pull request by aaannz**
- append (cancelled) when job is cancelled from test list
- redirect to new test if test is restarted from result view
- use $job->can_be_duplicated to check if even to display restart icon
- use test name as link even for scheduled tests

There is things I'm not happy about:
- I had a problem with firing double events for restart link until I stopped using $(document).on(..) and moved to $('#restart-result').click. This however needs to be called when restart-result id is know so I added it directly to result.html.ep. I don't think this is right approach.

### CODE REVIEW

File: public/javascripts/tests.js [hunk 1]
```diff
@@ -197,12 +197,21 @@ function renderTestsList(jobs) {
...
+});
+
+$(document).on('click', '.cancel', function() {
+    var cancel_link = $(this);
+    var test = $(this).parent('td');
+    $.post(cancel_link.attr("href")).done( function( data ) { $(test).append(' (cancelled)'); });
+    $(this).html('');
+});
+
+$(document).on('mouseover', '.parent_child', highlightJobs);
```
Comment (coolo) [inline public/javascripts/tests.js]: why did you move this out of the function? tests.js is included in all pages, so this code will run on all - but we have no parents/childs in other pages but tests, no?

File: templates/test/running_table.html.ep [hunk 2]
```diff
@@ -49,15 +49,17 @@
...
-		<td class="test">
-		    %= link_to url_for('test', 'testid' => $job->id) => begin
-                    <i class="status state_running fa fa-circle" title="Running"></i>
-		    %= link_post url_for('apiv1_cancel', 'name' => $job->id) => ('data-remote' => 'true') => (class => 'cancel') => begin
-		      <i class="action fa fa-times-circle-o"></i>
-		    % end
-                    %= $testname
-                    % end
-		</td>
+        <td class="test">
```
Comment (coolo) [inline templates/test/running_table.html.ep]: what happened to the indenting? used tabs?

Comment (coolo) [inline templates/test/running_table.html.ep]: looks like the original used tabs and your editor replaced them with 4 spaces

Comment (aaannz) [inline templates/test/running_table.html.ep]: Yes, I use spaces by default. Do we want tabs in html templates?

File: public/javascripts/tests.js (hunk 1 above)
Comment (aaannz) [inline public/javascripts/tests.js]: Probably during various experiments. I'll move it back.

File: templates/test/running_table.html.ep (hunk 2 above)
Comment (coolo) [inline templates/test/running_table.html.ep]: no, I want consistent indending on my editor (using tabs=8 :)

**Comment by coolo** [comment]
I will apply the fixes in another PR
